fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java

```diff
@@ -134,26 +135,35 @@ private void registerPeriodicalTask() {
             if (taskSchedule == null) {
                 continue;
             }
-
+            long period = TimeUtils.convertTimeUnitValueToSecond(taskSchedule.getPeriod(),
+                    taskSchedule.getTimeUnit());
             LocalDateTime startTime = Utils.getDatetimeFromLong(taskSchedule.getStartTime());
-            Duration duration = Duration.between(LocalDateTime.now(), startTime);
-            long initialDelay = duration.getSeconds();
-            // if startTime < now, start scheduling now
-            if (initialDelay < 0) {
-                initialDelay = 0;
-            }
+            LocalDateTime scheduleTime = LocalDateTime.now();
+            long initialDelay = getInitialDelayTime(period, startTime, scheduleTime);
             // Tasks that run automatically have the lowest priority,
             // but are automatically merged if they are found to be merge-able.
             ExecuteOption option = new ExecuteOption(Constants.TaskRunPriority.LOWEST.value(),
                     true, task.getProperties());
             ScheduledFuture<?> future = periodScheduler.scheduleAtFixedRate(() ->
                             executeTask(task.getName(), option), initialDelay,
-                    TimeUtils.convertTimeUnitValueToSecond(taskSchedule.getPeriod(),
-                            taskSchedule.getTimeUnit()), TimeUnit.SECONDS);
+                    period, TimeUnit.SECONDS);
             periodFutureMap.put(task.getId(), future);
         }
     }
 
+    @VisibleForTesting
+    static long getInitialDelayTime(long period, LocalDateTime startTime,
+                                    LocalDateTime scheduleTime) {
+        Duration duration = Duration.between(scheduleTime, startTime);
+        long initialDelay = duration.getSeconds();
+        // if startTime < now, start scheduling from the next period
+        if (initialDelay < 0) {
+            return ((initialDelay % period) + period) % period;
```

Comment:
is it equal to (initialDelay + period) % period?

Reply:
not exactly. eg: initialDelay=-200, period = 60:
```

((initialDelay % period) + period) % period = (-20 + 60) % 60 = 40;

(initialDelay + period) % period = -120 % 60  = 0
```

Reply:
-140 % 60  = 40

Reply:
in java. This result is -20